initial commit

viktorsohin opened the pull request:
added tests and instructions for running them

batmanastasiya commented on README.md:
````diff
@@ -1 +1,40 @@
+Install project dependencies and Playwright:
+
+```bash
+npm init playwright@latest
````
It seems that here should be just `npm i`
Also If you add account creation. There shoud be instruction where credentials should be added in the repository

batmanastasiya commented on package.json:
```diff
@@ -0,0 +1,26 @@
+  "scripts": {
+    "test-1": "npx playwright test tests/test-1.spec.ts",
+    "test-2": "npx playwright test tests/test-2.spec.ts",
+    "test-3": "npx playwright test tests/test-3.spec.ts"
```
Please add test:all + appropriate script OR/AND add it to the readme

batmanastasiya commented on tests/test-1.spec.ts:
```diff
@@ -0,0 +1,18 @@
+
+test('Verify login with valid credentials', async ({ page }) => {
+
+  await page.goto('https://practicesoftwaretesting.com/auth/login');
```
Please, add URLs to the separate file and use it through the tests OR consider using `.env` file

batmanastasiya commented on tests/test-1.spec.ts:
```diff
@@ -0,0 +1,18 @@
+  await page.goto('https://practicesoftwaretesting.com/auth/login');
+
+  await page.locator('#email').fill('customer@practicesoftwaretesting.com');
+  await page.locator('#password').fill('welcome01');
```
Same for the credentials

batmanastasiya commented on tests/test-1.spec.ts:
```diff
@@ -0,0 +1,18 @@
+
+  const usernameLocator = page.locator('nav').getByText('Jane Doe');
+  await expect(usernameLocator).toBeVisible();
+});
```
All the element listed in this file can be foung by test id. Please, chnge all `page.locator(...)` to `page.getByTestId(...)`
https://playwright.dev/docs/api/class-framelocator#frame-locator-get-by-test-id

batmanastasiya replied:
Test id attribute should be added to the playwright.config 
https://playwright.dev/docs/api/class-testoptions#test-options-test-id-attribute

batmanastasiya commented on tests/test-3.spec.ts:
```diff
@@ -0,0 +1,36 @@
+    await expect(alert).toBeVisible();
+    await expect(alert).toHaveText('Product added to shopping cart.');
+
+    await page.waitForTimeout(8000);
```
Please, do not use waiters like that. It is an explicit wait. Bad practice. Please, try to find another solution

batmanastasiya reviewed (changes requested):
Hey! Looks nice! Added a few comments to improve solution a little bit

Also, there is no need to add extra space between each and every expect. It is great when we have visual division between actions and expects. But usually they go like blocks. Block of actions/ block of assertions

viktorsohin commented:
the fix includes:
- updated the README.md file
- all tests can be run with a single command
- element selection is now done using built-in Playwright locators
- sensitive data moved to a separate credentials.ts file
- implemented a new solution for checking an ‘Product added to shopping cart’ alert with a timer
- removed unnecessary spacing between lines of test code

EugenePetrik commented on tests/test-1.spec.ts:
```diff
@@ -0,0 +1,13 @@
+import { credentials } from '../credentials.ts'; 
+
+test('Verify login with valid credentials', async ({ page }) => {
+  await page.goto(credentials.loginPage);
```
I'd add baseURL in https://github.com/viktorsohin/playwright_fnl_project/pull/1/files#diff-f679bf1e58e8dddfc6cff0fa37c8e755c8d2cfc9e6b5dc5520a5800beba59a92R30

EugenePetrik reviewed (approved):
Good job! 👍 Approved but left a small comment
